
syntax = "proto3";

package bidirectional;

service Bidirectional {
   rpc ServerSend(Text) returns (Res) {}
   rpc ChangeAccountState(Account) returns (Res) {}
   rpc ClientStream(Empty) returns (stream Text) {}
}

message Text {
  string sender = 1; 
  string receiver = 2; 
  string message = 3;
}

message Account { 
  int32 type = 1; 
  string username = 1; 
}

message Res { 
  int32 status = 1;
}

message Empty {}